docs/adr/0001-daily-active-use-server-side-metrics-glean.md

```diff
@@ -0,0 +1,121 @@
+
+## Pros and Cons of the Options
+
+### A. Glean Parser for Rust
+
+Glean is a widely used tool at Mozilla and provides us with a solution to the given issue and possible extensibility in the future. Not without some challenges in initial implementation, related to coordination with Glean team and upfront development effort. However, the potential for positive impact within our team and the organization is significant: all Rust server applications will be able to use Glean with full server support, our possible intention to integrate Glean into Push is made easier, and this is done in partnership with the Glean team.
+
+#### Pros
+
+* Makes Glean compatible for all Rust server applications going forward.
+* Preferred option of the Glean team.
+* Glean team believes, based on FxA metrics volume, that our volume will not be a problem (180-190K per minute).
+* A collection of metrics, emitted as a single "Ping Event" make querying of related data simpler.
+* Core of Glean's purpose is to measure user interactions and the rich metadata that accompanies it.
+* Capacity for future expansion of application metrics within Sync, beyond DAU.
+* Prepares for implementation of same measurements in autopush, also using Glean.
+* Easier to query, have data team support to set up queries.
+* Use of standardized Mozilla tooling.
+* Establishment of team knowledge of using Glean.
+* Establishment of server-side Rust best practices, leading to easier development for backend Rust applications.
+* Transparency in data review process with consideration made to minimum collection of data.
+
+#### Cons
+
+* The `glean_parser` tool used by other Glean applications is currently not supported for Rust. Furthermore, client applications can use the Glean SDK and this is also not an option for us.
+* Server side metrics have not yet been implemented for a Rust server application of this kind, so this is new territory.
+* There is added complexity of data review process and registration of the application to Glean's probe scraper, PubSub topics, ingestion pipeline etc.
+* May require more uptfront development effort to get up to speed with new code and process.
+* Some concerns exist around volume of data emitted from the service and if it is feasible, but we won't know until we try.
+
+### B. Custom Glean Implementation 
+
+This was originally appearing to be the desired approach to measuring DAU via Glean.  This was predicated on the ease of prototyping a custom implementation that imitated the Glean team's logic to create "Glean-compliant" output that could be configured to be ingested.  However, in consulting with the Glean team and evaluating the pros and cons of this approach, it became clear this approach had considerably more drawbacks than implementing the `glean_parser` for Rust. These drawbacks were a lack of testing, validation, less support from the Glean team, and the potential problems with maintenance and adding Glean metrics in the future.
```

Comment:
nit: this is a bit verbose for what it's trying to do, but I'm not gonna hold up the ADR on a revision of it.

Reply:
I tightened it up a bit nonetheless, good point